### DISCUSSION TITLE
Why should `defaultValues` require a value?

### DISCUSSION THREAD

**Discussion by GollyJer** [react-hook-form/react-hook-form#4300] [Ideas] [answered]
Maybe this belongs in the `V7` discussion, but putting it here for now....

In this example `me` comes from an async query so `me.firstName` is undefined on the first render.

Why should this work?

```javascript 
const defaultValues = useMemo(
  () => ({
    first_name: me?.firstName || null,
  }),
  [me?.firstName]
);
```

But this throws the error.
```first_name is missing in the 'defaultValue' prop of either its Controller (https://react-hook-form.com/api#Controller) or useForm (https://react-hook-form.com/api#useForm)```
```javascript 
const defaultValues = useMemo(
  () => ({
    first_name: me?.firstName,
  }),
  [me?.firstName]
);
```

It seems unnecessary to require a value.

**Comment by bluebill1049** [comment, marked as answer]
hook form support inline defaultValue and also an uncontrolled component, so if nothing is defined at `defautlValue` then we will look up for `defaultValues` as `useform`, also working with FieldArray is required to give a `defaultValue` when inputs shuffle around.
